js/bundle.go

```diff
@@ -436,6 +436,20 @@ func (b *Bundle) setInitGlobals(rt *sobek.Runtime, vu *moduleVUImpl, modSys *mod
 		}
 		return openImpl(rt, b.filesystems["file"], pwd, filename, args...)
 	})
+	warnAboutModuleMixing := func(name string) {
+		warnFunc := rt.ToValue(func() error {
+			return fmt.Errorf(
+				"you are trying to access identifier %q, this likely is due to mixing ESM and CommonJS syntax."+
+					"This isn't supported in the JavaScript standard, please use only one or the other",
```

Comment:
> It tried to access the _module_ identifier, this operation is not supported by JavaScript ECMAScript standard. It happens if the script mixes CommonJS and ECMAScript Modules syntaxes.

Suggesting an alternative, feel free to discard or change

Reply:
It is possible that it was coding mistake though, which is why I don't want to say it is definitely mixing.

And the standard lets you have idenfitiers that are `module` or `exports` - just in this case they weren't defined until the global object

Reply:
To be honest I am not particularly happy with the current wording either, just pointing out why it was choosen and worded in the particular way.

Reply:
What about this new suggestion?

>There was an attempt to access the _module_ identifier. It might be an issue generated by the script's code if it is mixing CommonJS and ECMAScript Modules syntaxes, which is not supported by JavaScript ECMAScript standard.

I'm using _module_ just as a placeholder, of course, we will have there a `%q``.

Reply:
JavaScript ECMAScript standard sounds ... bad to me, and "generated by the script's code":
1. generated is not the use I will use. Arguably this makes the whole sentence longer than "It might be an issue with code that mixes ...." 
2. script's code - technically here in one of the cases it isn't a script it is a module - that is what ESM denotes.

Other than that I am okay with both the current and your wording, and do not have a strong preference, even the my nitpicks are not very serious IMO.

I just would prefer for us to discuss it with the other reviewers before I make one more change so I stop changing the message :)

Reply:
@joanlopez @olegbespalov we need your input here :pray:

Reply:
I think the current version is good enough, except that I don't know how clear are the specific terms ESM, CommonJS, and JavaScript standard.

Reply:
I don't have a strong opinion, however I do believe that just ESM isn't enough, IMO we should use and refer to it as `ECMAScript Modules (ESM)`

Reply:
I  can change it ... but IMO if people don't know what ESM is in this context with CommonJS ... ECMAScript Modules will be just as hard to understand.

I am not against it, just please make a code suggestion.

Reply:
> if people don't know what ESM is in this context with CommonJS

That's the trick. If they don't know what it is ESM, they probably don't know ECMAScript Modules either, but the `ECMAScript Modules` are more searchable IMO

So my minimal requested change looks like this:
```suggestion
				"you are trying to access identifier %q, this likely is due to mixing ECMAScript Modules (ESM) and CommonJS syntax."+
					"This isn't supported in the JavaScript standard, please use only one or the other",
```

For the other changes in the sentence I don't have strong opinion :shrug: